### DISCUSSION TITLE
Gitlab refs for private repositories

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#12016] [Request Help]
Is it possible to get renovate to update gitlab refs for private gitlab enterprise repos?
Currently i'm using `gitlab-tags` as the datasource for what I need, but that would not include gitlab revisions.
I was hoping I could use `git-refs` but it does not seem to support authentication and I have a setup where I can't use SSH keys on the machine it runs.
Is ssh keys for `git-refs` the only way?

I apologize if this has been asked before, but there's a lot of gitlab mentions and I did not manage to find one that matches what I wanted to know.

**Comment by rarkins** [comment]
GitLab tags should query over https so should work if you configure a hostRules with token for it - have you tried?

**Reply by HenrikPoulsen** [reply]
Getting tags works fine. I use a hostrule and then use my gitlab token as the token for it.
The problem isn't the tags. The problems is that `gitlab-tags` doesn't support commit shas (as in if someone uses the sha as the "version" it wont discover new shas).
`git-refs` does but doesn't seem to support hostRules so I can't set a custom auth token for those requests, and for security reasons I can't use ssh keys for it.
So what I want is to update one gitlab commit sha from the main branch to the latest commit sha on the main branch. In a private gitlab enterprise repo that I need to use an access token for

**Reply by rarkins** [reply]
I checked and it doesn't look like we satisfy this for GitHub either?

Maybe we can use a git "insteadOf" directive to redirect it to https using the token?

**Reply by HenrikPoulsen** [reply]
Thanks I'll give it a shot
